packages/editor/src/components/post-actions/actions.js

```diff
@@ -663,6 +664,116 @@ const renamePostAction = {
 	},
 };
 
+function ReorderModal( { items, closeModal, onActionPerformed } ) {
+	const [ item ] = items;
+	const { editEntityRecord, saveEditedEntityRecord } =
+		useDispatch( coreStore );
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+	const [ orderInput, setOrderInput ] = useState( item.menu_order );
+
+	async function onReorder( event ) {
+		event.preventDefault();
+		if (
+			! Number.isInteger( Number( orderInput ) ) ||
+			orderInput?.trim?.() === ''
+		) {
+			return;
+		}
+		try {
+			await editEntityRecord( 'postType', item.type, item.id, {
+				menu_order: orderInput,
+			} );
+			closeModal();
+			// Persist edited entity.
+			await saveEditedEntityRecord( 'postType', item.type, item.id, {
+				throwOnError: true,
+			} );
+			createSuccessNotice( __( 'Order updated' ), {
+				type: 'snackbar',
+			} );
+			onActionPerformed?.( items );
+		} catch ( error ) {
+			const errorMessage =
+				error.message && error.code !== 'unknown_error'
+					? error.message
+					: __( 'An error occurred while updating the order' );
+			createErrorNotice( errorMessage, {
+				type: 'snackbar',
+			} );
+		}
+	}
+	const saveIsDisabled =
+		! Number.isInteger( Number( orderInput ) ) ||
+		orderInput?.trim?.() === '';
+	return (
+		<form onSubmit={ onReorder }>
+			<VStack
+				spacing="5"
+				style={ {
+					// editor styles are not loaded in dataviews so we use an inline style here.
+					maxWidth: 384 - 32 - 32 - 6 - 6,
+				} }
+			>
+				<div>
+					{ __(
+						'Determines the order of pages in the Page List block. Pages with the same order value are sorted alphabetically. Negative order values are supported.'
+					) }
```

Comment:
```suggestion
					{ __(
						'Determines the order of pages. Pages with the same order value are sorted alphabetically. Negative order values are supported.'
					) }
```

I think the order affects not only the Page List block, but also the order of pages on the dashboard.

So, how about changing it like this?

Reply:
> but also the order of pages on the dashboard.

Hmm, where do you mean? I think it's helpful to be explicit here... "The order of pages." is a bit ambiguous.

Reply:
Sorry, my explanation was insufficient. I wanted to mention the following screen.

![image](https://github.com/WordPress/gutenberg/assets/54422211/36a33c4e-afa9-43ef-b4b7-5cdc3bc97688)

In the screenshot above, the order of "Page 2" has been changed to 1.

Reply:
Oh yes, I see. I wonder if that's intentional because if you refresh then the `order` is no longer respected 🤔  Sorting by date (etc.) also ignores `order`.

Reply:
The default sorting takes order into consideration, but if we sort by any field order is not taken into account.

Reply:
I applied the suggestion.